ui/app/controllers/pipelines.js

```diff
@@ -25,7 +25,11 @@ export default class PipelinesController extends Controller {
 
   @action
   async destroyPipeline(pipeline) {
-    await pipeline.destroyRecord();
+    try {
+      await pipeline.destroyRecord();
+    } catch (e) {
```

Comment:
Nice catch! I vaguely remember us having to guard check against a similar issue on plaform-ui before, so this is great!